src/distilabel/llm/litellm.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import PrivateAttr
+
+from distilabel.llm.base import AsyncLLM
+
+if TYPE_CHECKING:
+    from distilabel.llm.typing import GenerateOutput
+    from distilabel.steps.task.typing import ChatType
+
+
+class LitellmLLM(AsyncLLM):
```

Comment:
To me `LitellmLLM` is redundant, so I'd go with something like `LiteLLM` and that's it, similarly to what we did with `vLLM`

Reply:
```suggestion
class LiteLLM(AsyncLLM):
```